File: 1915-check-if-one-string-swap-can-make-strings-equal/1915-check-if-one-string-swap-can-make-strings-equal.java
class Solution {
    public boolean areAlmostEqual(String s1, String s2) {
        int c =0;
        char[] ch1 = new char[2];
        char[] ch2 = new char[2];
        for(int i =0;i<s1.length();i++){
            if(s1.charAt(i)!=s2.charAt(i)){
                if(c==2) return false;
                ch1[c]=s1.charAt(i);
                ch2[c]=s2.charAt(i);
                c++;
            }
        }
        if(c==1) return false;
        Arrays.sort(ch1);Arrays.sort(ch2);
        if(ch1[0]!=ch2[0] || ch1[1]!=ch2[1]) return false;
        return true;
    }
}

File: 1915-check-if-one-string-swap-can-make-strings-equal/1915-check-if-one-string-swap-can-make-strings-equal.cpp
class Solution {
public:
    bool areAlmostEqual(string s1, string s2) {
        vector<int>ind;
        int n=s1.size();
        for(int i=0;i<n;i++){
            if(ind.size()>2) return false;
            if(s1[i]!=s2[i]) ind.push_back(i);
        }
        if(ind.empty()) return true;
        if(ind.size()>2 or ind.size()==1) return false;
        swap(s1[ind[0]],s1[ind[1]]);
        if(s1==s2) return true;
        swap(s1[ind[0]],s1[ind[1]]);
        swap(s2[ind[0]],s2[ind[1]]);
        return (s1==s2);
    }
};